### ISSUE TITLE
iconv(1) self license violation about BSDL 2-clause.

### ISSUE THREAD

**Issue by ikazont** [ikazont/netbsd#17]
**[Original report](https://bitbucket.org/tnozaki/netbsd/issues/17) by Takehiko NOZAKI (Bitbucket: [tnozaki](https://bitbucket.org/tnozaki), GitHub: [ikazont](https://github.com/ikazont)).**

----------------------------------------

BSDL includes following words:

> 2. Redistributions in binary form must reproduce the above copyright notice, ...

this means, copyright holder embed the copyright string into binary, and can reproduce it by what(1) such as:

```
#!sh

$ what /usr/bin/iconv
iconv
         Copyright (c) 2003, 2007, 2013, 2014 Citrus Project, All rights reserved.

```

but iconv.c doesn't embeded copyright string by __COPYRIGHT() macro.

**Comment by ikazont** [comment]
**Original comment by Takehiko NOZAKI (Bitbucket: [tnozaki](https://bitbucket.org/tnozaki), GitHub: [ikazont](https://github.com/ikazont)).**

----------------------------------------

BUGFIX: Issue #17 -- iconv(1) self license violation about BSDL 2-clause.
update (c) year and embed (c) in binary by __COPYRIGHT() macro.

**Comment by ikazont** [comment]
**Original comment by Takehiko NOZAKI (Bitbucket: [tnozaki](https://bitbucket.org/tnozaki), GitHub: [ikazont](https://github.com/ikazont)).**

----------------------------------------

Squashed commit of the following:

commit 5a766e5a96c5deafe081f180bb551022432821af
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 17:01:02 2014 +0900

    BUGFIX: small fix related <<cset 1bd468ad7da163fdbcea15cddfe469d566ca72e7>>, pointer 'in' should be updated.

commit 805495a2ec89e80c5d2391692ff98557cb921ae1
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 16:56:59 2014 +0900

    CLEANUP: check fwrite error strictly.

commit 8aaac3208aade9a4990643ed69423e076eeabd5c
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 16:51:12 2014 +0900

    CLEANUP: wrap lines longer than 80 characters.

commit e05b9f50f91114211a2525a17d236d25c0da8576
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 16:46:27 2014 +0900

    BUGFIX: Issue #17 -- iconv(1) self license violation about BSDL 2-clause.
    update (c) year and embed (c) in binary by __COPYRIGHT() macro.

commit 742f78e6003d6e7e15f2f39a0f854513e1642ea5
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 16:32:55 2014 +0900

    BUGFIX: Issue #16 -- iconv(1) option "-l" + "-c" are not allowed.

commit 901f6b6126f081b0c41283b19e8304d34d5e92df
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 16:23:59 2014 +0900

    CLEANUP: use reader/writer instread of fp/stdin/stdout for code readability.

commit e6fb4e8725459479bd2b317a70c8947dbf62d035
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 16:22:45 2014 +0900

    CLEANUP: use EXIT_FAILURE instead of 1.

commit ec79a546afabe793900b5a4e2e1bddb2350ef174
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 16:15:50 2014 +0900

    CLEANUP: use stdio BUFSIZ instead of INBUFSIZE/OUTBUFSIZE.
    there is no reason for OUTBUFSIZ$ to allocate twice size of INBUFSIZE.

commit 88361b1803f68638661ef5c094c240452eb176fa
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 16:11:59 2014 +0900

    CLEANUP: change variable name opt_f/opt_t -> from/to for code readability.

commit d0d2b01be6c7b5eb59344e0842a42db6a73614e7
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 16:08:52 2014 +0900

    CLEANUP: change variable name opt_l -> list for code readability.

commit eb336d704b4dcc62a468315b68cee012541eda18
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 16:06:50 2014 +0900

    CLEANUP: move "flags" switch into converter_t.

commit 4bef0fb0ef22f9e6b341bf6da0abbbabf67f554b
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 16:04:51 2014 +0900

    CLEANUP: move "silent" switch into converter_t.

commit 8db0323753b7a19a4facbdbdcc5675295ce42910
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 16:01:45 2014 +0900

    CLEANUP: add inbuf/outbuf to converter_t, now input/output buffer may be reused.

commit e9c69fa252a6336316478416e798ae34b383020e
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 15:59:28 2014 +0900

    CLEANUP: introduce converter_t, wrapper for iconv_t and buffers, flags and so on(currently only holds iconv_t).

commit 1bd468ad7da163fdbcea15cddfe469d566ca72e7
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 15:49:39 2014 +0900

    BUGFIX: Issue #15 -- iconv input/output buffer should be dynamically allocated for M:N conversion.
    * N:1 conversion may exhaust input buffer, EINVAL case and not input buffer
      consumpution need to increase the buffer.
    * 1:N conversion with irreversible chars(such as GNU libiconv //TRANSLIT)
      may exhaust output buffer, so E2BIG case need to increase the buffer.

commit bb23f038f6f2d6a8757fc975d8a0b875c2d9fee9
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 15:30:10 2014 +0900

    CLEANUP: sortify(out -> outbytes order) for code readability.

commit 8ad7faa4c36eed432a440a3ac20585ca001b0a10
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 15:28:48 2014 +0900

    CLEANUP: use goto label for after unexpected eof error handling.

commit 62b32326332d248d55b2011cb7307273e822dd21
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 15:27:08 2014 +0900

    CLEANUP: remove useless else clause.

commit dfa7d0579e2ba5c7a5ffed834ed22020d1634a8e
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 15:25:40 2014 +0900

    CLEANUP: use goto label for generic iconv error.

commit 756f7dd7c14c2f1300c0b773f776d47d1a8ce703
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 15:22:34 2014 +0900

    CLEANUP: store converted size to variable for code readability.

commit 174f687fb76168a83904732d102e49e7caa44eb9
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 15:19:02 2014 +0900

    CLEANUP: fix comment position and grammer, no functional change.

commit 902fc9205077ffb9ac9bd0e665903eeb7e071d01
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 15:18:10 2014 +0900

    CLEANUP: use "%zu" format string for print size_t instead of cast to unsigned long.

commit 1e2f4fb607cc2a9316cbf44ad35cc8b12649df56
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 15:15:30 2014 +0900

    CLEANUP: don't reuse same variable ret for code readability.

commit 9012cc498c05ffbb5e91f3de0d334143676a68ef
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 15:13:09 2014 +0900

    CLEANUP: use puts() instead of printf("%s\n") for lazy compiler lacks optimization.

commit a7788b8fd7d61b20049bd5921bf3ab417fd65fe2
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 15:12:07 2014 +0900

    CLEANUP: change syntax, no functional change.

commit 9a592a01f0005b3519f74eeae5e7c91371deddb0
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 15:10:45 2014 +0900

    CLEANUP: more change do_conv() interface, output stream should be supplied from outer function.

commit 66cd6d3417fea575580a766c64beaf9f2e9c223c
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 15:07:17 2014 +0900

    CLEANUP: more change do_conv() interface, remove unused argument of filename.

commit 90ef538d47adf05f41269357b1861c47ffd1c8ba
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 15:05:06 2014 +0900

    CLEANUP: more change do_conv() interface, iconv_t should be reused and supplied from outer function.

commit 5c39be4c40f209ff4b47d537232336579461efb9
Author: Takehiko NOZAKI <tnozaki@outlook.com>
Date:   Sun Jul 20 15:02:48 2014 +0900

    CLEANUP: change do_conv() interface, flags for __iconv() should be supplied from outer funcion.
